src/Nncase.Tests/Evaluator/UnitTestEvaluatorMath.cs

```diff
@@ -0,0 +1,1033 @@
+
+            // if (op != BinaryOp.LogicalAnd && op != BinaryOp.LogicalOr && op != BinaryOp.LogicalXor)
+            if (op == BinaryOp.LeftShift || op == BinaryOp.RightShift)
+            {
+                TestBinaryRunNormal(op, OrtKISharp.Tensor.MakeTensor(a, new long[] { 2, 4 }), OrtKISharp.Tensor.MakeTensor(b, new long[] { 2, 4 }),
+                    Tensor.From(a, new[] { 2, 4 }), Tensor.From(b, new[] { 2, 4 }));
+            }
+        }
+
+        // float
+        foreach (var op in ops)
+        {
+            var a = new float[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+            var b = new float[] { 1, 1, 2, 2, 3, 3, 4, 4 };
+            if (op != BinaryOp.LogicalAnd && op != BinaryOp.LogicalOr && op != BinaryOp.LogicalXor && op != BinaryOp.LeftShift && op != BinaryOp.RightShift)
+            {
+                TestBinaryRunNormal(op, OrtKISharp.Tensor.MakeTensor(a, new long[] { 2, 4 }), OrtKISharp.Tensor.MakeTensor(b, new long[] { 2, 4 }),
+                    Tensor.From(a, new[] { 2, 4 }), Tensor.From(b, new[] { 2, 4 }));
+            }
+        }
+
+        // int
+        foreach (var op in ops)
+        {
+            var a = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+            var b = new int[] { 1, 1, 2, 2, 3, 3, 4, 4 };
+            if (op != BinaryOp.LogicalAnd && op != BinaryOp.LogicalOr && op != BinaryOp.LogicalXor && op != BinaryOp.LeftShift && op != BinaryOp.RightShift)
+            {
+                TestBinaryRunNormal(op, OrtKISharp.Tensor.MakeTensor(a, new long[] { 2, 4 }), OrtKISharp.Tensor.MakeTensor(b, new long[] { 2, 4 }),
+                    Tensor.From(a, new[] { 2, 4 }), Tensor.From(b, new[] { 2, 4 }));
+            }
+        }
+
+        // long
+        foreach (var op in ops)
+        {
+            var a = new long[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+            var b = new long[] { 1, 1, 2, 2, 3, 3, 4, 4 };
+            if (op != BinaryOp.LogicalAnd && op != BinaryOp.LogicalOr && op != BinaryOp.LogicalXor && op != BinaryOp.LeftShift && op != BinaryOp.RightShift)
+            {
+                TestBinaryRunNormal(op, OrtKISharp.Tensor.MakeTensor(a, new long[] { 2, 4 }), OrtKISharp.Tensor.MakeTensor(b, new long[] { 2, 4 }),
+                    Tensor.From(a, new[] { 2, 4 }), Tensor.From(b, new[] { 2, 4 }));
+            }
+        }
+    }
+
+    [Fact]
+    public void TestBinaryRunInvalidType()
+    {
+        {
+            var ops = new BinaryOp[] { BinaryOp.LogicalAnd, BinaryOp.LogicalOr, BinaryOp.LogicalXor, BinaryOp.LeftShift, BinaryOp.RightShift };
+            foreach (var op in ops)
+            {
+                var expr = IR.F.Math.Binary(op, 1f, 2f);
+                CompilerServices.InferenceType(expr);
+                Assert.IsType<InvalidType>(expr.CheckedType);
+            }
+        }
+    }
+
+    private void TestBinaryRunNormal(BinaryOp op, OrtKISharp.Tensor ort_a, OrtKISharp.Tensor ort_b, Expr exp_a, Expr exp_b)
+    {
+        OrtKISharp.Tensor expect;
+        switch (op)
```

Comment:
it is recommended to use switch expr instead of switch statement when returning different values for different inputs, this will make the code more concise